compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -677,13 +677,24 @@ where
                         });
                     }
 
-                    match tcx.struct_tail_erasing_lifetimes(pointee, cx.param_env()).kind() {
-                        ty::Slice(_) | ty::Str => TyMaybeWithLayout::Ty(tcx.types.usize),
-                        ty::Dynamic(_, _, ty::Dyn) => {
-                            TyMaybeWithLayout::Ty(tcx.mk_imm_ref(
+                    let metadata = if let Some(metadata_def_id) = tcx.lang_items().metadata_type() {
+                        let metadata = tcx.normalize_erasing_regions(
+                            cx.param_env(),
+                            tcx.mk_projection(metadata_def_id, tcx.mk_substs_trait(pointee, &[])),
+                        );
+
+                        // Map `Metadata = Dyn<dyn Trait>` back to a vtable, since it
```

Comment:
What is `Dyn<dyn Trait>`? Should that be [`DynMetadata<dyn Trait>`](https://doc.rust-lang.org/std/ptr/struct.DynMetadata.html)?

Reply:
ah yes, typo